fs/tmpfs/fs_tmpfs.c

```diff
@@ -55,27 +55,12 @@
 #  warning CONFIG_FS_TMPFS_FILE_FREEGUARD needs to be > ALLOCGUARD
 #endif
 
-#define tmpfs_lock_file(tfo) \
-           (tmpfs_lock_object((FAR struct tmpfs_object_s *)tfo))
-#define tmpfs_lock_directory(tdo) \
-           (tmpfs_lock_object((FAR struct tmpfs_object_s *)tdo))
-#define tmpfs_unlock_file(tfo) \
-           (tmpfs_unlock_object((FAR struct tmpfs_object_s *)tfo))
-#define tmpfs_unlock_directory(tdo) \
-           (tmpfs_unlock_object((FAR struct tmpfs_object_s *)tdo))
-
```

Comment:
It's better to keep these macros:
```
#define tmpfs_lock(fs) \
           nxrmutex_lock(&fs->tfs_lock)
#define tmpfs_lock_file(tfo) \
           nxrmutex_lock(&tfo->tfo_lock)
#define tmpfs_lock_directory(tdo) \
           nxrmutex_lock(&tdo->tdo_lock)
#define tmpfs_unlock(fs) \
           nxrmutex_unlock(&fs->tfs_lock)
#define tmpfs_unlock_file(tfo) \
           nxrmutex_unlock(&tfo->tfo_lock)
#define tmpfs_unlock_directory(tdo) \
           nxrmutex_unlock(&tdo->tdo_lock)
```
to avoid the cast spread in the whole source code.